### PR TITLE
Feature/smos l2nrt ssm

### LINKED ISSUE #408: Work on the initial implementation for SMOS Level 2 NRT Soil Moisture ioda-converter
SMOS L2 NRT soil moisture website: 
https://earth.esa.int/eogateway/missions/smos
https://smos-diss.eo.esa.int/oads/access/

Data file example: 
W_XX-ESA,SMOS,NRTNN_C_LEMM_20201110022133_20201109235712_20201110012802_o_v200_l2sm.nc

Documents and user handbook: 
https://earth.esa.int/documents/10174/1854503/SMOS-level-2-Soil-Moisture-NRT-V200-release-note

Related (Key) Documentation: 
See Related (Key) Documentation  tab in https://earth.esa.int/web/guest/missions/esa-operational-eo-missions/smos/content/-/asset_publisher/t5Py/content/data-types-levels-formats-7631

Pre-QC uses RFI probability flag as suggested by Kerr et al. (2016 and Al-Yaari et al. (2017): If RFI_probability >20, soil moisture data quality is poor. Otherwise, soil moisture data quality is good.

References:

A. Al-Yaari, J.-P. Wigneron, Y. Kerr, N. Rodriguez-Fernandez, P.E. O'Neill, T.J. Jackson, G.J.M. De Lannoy, A. Al Bitar, A. Mialon, P. Richaume, J.P. Walker, A. Mahmoodi, S. Yueh, 2017: Evaluating soil moisture retrievals from ESA's SMOS and NASA's SMAP brightness temperature datasets, Remote Sensing of Environment, 93, 257-273, https://doi.org/10.1016/j.rse.2017.03.010.

Kerr YH, Al-Yaari A, Rodriguez-Fernandez N, Parrens M, Molero B, Leroux D, Bircher S, Mahmoodi A, Mialon A, Richaume P, Delwart S, Al Bitar A, Pellarin T, Bindlish R, Jackson TJ, Rüdiger C, Waldteufel P, Mecklenburg S, Wigneron JP, 2016: Overview of SMOS performance in terms of global soil moisture monitoring after six years in operation, Remote Sensing of Environment, 180, 40-63.

Oliva R, Daganzo E, Kerr YH, Mecklenburg S, Nieto S, Richaume P, Gruhier C. SMOS Radio Frequency Interference Scenario: Status and Actions Taken to Improve the RFI Environment in the 14001427-MHz Passive Band, 2012: IEEE Transactions on Geoscience and  Remote Sensing, 50, 1427-1439. 

The purpose of this initial implementation is to support the community land DA task.

### PR DESCRIPTION
**Pull request by YoulongXia-NOAA**
## Description

Initial Implementation for SMOS Level 2 NRT Soil Moisture

### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #408 

## Dependencies

No dependencies

## Impact

No

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -598,3 +600,13 @@ ecbuild_add_test( TARGET  test_${PROJECT_NAME}_smap_ssm
                             -o testrun/smap_ssm.nc
                             -m maskout"
                             smap_ssm.nc ${IODA_CONV_COMP_TOL_ZERO})
+
+ecbuild_add_test( TARGET  test_${PROJECT_NAME}_smos_ssm
+                    TYPE    SCRIPT
+                    COMMAND ${CMAKE_BINARY_DIR}/bin/iodaconv_comp.sh
+                    ARGS    netcdf
+                            "${CMAKE_BINARY_DIR}/bin/smos_ssm2ioda.py
```
Comment (aerorahul) [inline test/CMakeLists.txt:608]: This needs to be PROJECT_BINARY_DIR

**Review by aerorahul — CHANGES_REQUESTED** [review]

**Comment by YoulongXia-NOAA** [comment]
Dear my colleagues,

I have developed a converter for SMOS Level 2 NRT soil moisture in support JEDI land DA activity. When your time is available, please review it. Thank you.

**Review by aerorahul — APPROVED** [review]

**Review by CoryMartin-NOAA — APPROVED** [review]

**Review by srherbener — APPROVED** [review]
Looks good. Thanks for contributing this script.

**Comment by YoulongXia-NOAA** [comment]
Thank you all for your review. This work is done now.  Have a good day and weekend.

**Pull request merged**